src/types/open-api.ts

```diff
@@ -61,6 +61,13 @@ export interface OpenAPIPath {
   $ref?: string;
 }
 
+export type OperationCustomBadgeType = 'primary' | 'success' | 'warning' | 'error' | 'experimental';
+
+export interface OperationCustomBadge {
```

Comment:
Went with `OperationCustomBadge` because `OperationBadge` was already a used name elsewhere in the code. Could probably land on a better name in the end 🤷‍♂️